apps/storybook/src/Selection.stories.tsx

```diff
@@ -52,19 +100,17 @@ const Template: Story<TemplateProps> = (args) => {
       >
         <PanMesh disabled={disablePan} />
         <ZoomMesh disabled={disableZoom} />
-        {selection === 'line' && (
-          <LineSelectionMesh
-            onSelection={(start, end) => setSelectedVectors([start, end])}
-            modifierKey={modifierKey}
-            color={color}
-          />
-        )}
-        {selection === 'rectangle' && (
-          <RectSelectionMesh
-            onSelection={(start, end) => setSelectedVectors([start, end])}
+        {selection && (
+          <DataSelectionMesh
+            onSelectionStart={(start) => {
+              setEnd(undefined);
+              setStart(start);
+            }}
+            onSelectionEnd={setEnd}
```

Comment:
Following on my comment in story `PersistSelection`, the API I'm suggesting is not sufficient to implement this story. The goal here is to display the selection coordinates as the user is drawing the selection (i.e. from `onPointerMove`), so we need a third callback, perhaps `onSelectionChange(selection: Selection) => void`. The callbacks would then be used as follows:

```tsx
onSelectionChange={setSelection}
onSelectionEnd={() => setSelection(undefined)}
```